evmlib/src/testnet.rs

```diff
@@ -58,6 +58,7 @@ pub fn start_node() -> AnvilInstance {
     // Spin up a local Anvil node.
     // Requires you to have Foundry installed: https://book.getfoundry.sh/getting-started/installation
     Anvil::new()
+        .port(4343_u16)
```

Comment:
I think the behaviour should be here that we can optionally specify a port or else it should be random. If we use a constant port and have tests running on the same machine, they might yield "address already in use"

Reply:
Shouldn't we run a single anvil node usually? I don't think we will run two local testnets at the same time? I think this is the same with other ports we have used in the past like `8000` for the faucet?

Reply:
I think that every test should start a new EVM node, just to have a clean starting state and to avoid side effects.

Reply:
In that case, we could still run it on the same port as we have only a single evm node running for each test?

Reply:
Tests run in parallel if the host machine is multi-threaded. And also, the OS can sometimes take a while to free a port after closing an application. That's what I'm a bit worried about.

We can always just try this out though and fix later if we run into problems. Having to manually specify all the EVM params is annoying.